src/TestProblem/Hydro/KelvinHelmholtzInstability/Init_TestProb_Hydro_KelvinHelmholtzInstability.cpp

```diff
@@ -122,9 +122,24 @@ void SetParameter()
 
    delete ReadPara;
 
-// set the random seed of each MPI rank
-   srand48_r( KH_RSeed+MPI_Rank, &drand_buf );
+// get the number of OpenMP threads
+   int NT;
+#ifdef OPENMP
+#pragma omp parallel
+#pragma omp master
+   {  NT = omp_get_num_threads();  }
+#else
+   {  NT = 1;                      }
+#endif
+
+//    allocate RNG
+   RNG = new RandomNumber_t( NT );
 
+// set the random seed of each MPI rank
+   for (int t=0; t<NT; t++) {
+      if ( KH_AllRankSame )   RNG->SetSeed(t, KH_RSeed+t);
```

Comment:
@hyschive this variable is not defined anywhere that I can see. I don't recall introducing it. Do you know where/how we should define it?

Reply:
Hmm it looks like I indeed do this. Ok, I need to figure out what I was trying to do here.

Reply:
I've removed the option `KH_AllRankSame` to simplify this test (so different MPI ranks will just have different random seeds). So no need to worry about it.

Reply:
Ok, sorry for the noise. I just pushed another commit to resolve this. I think this is ready to go now!